# Imports

In [1]:
import numpy as np
import pandas as pd

In [2]:
df_neurons = pd.read_csv('traced-neurons.csv')
df_neurons

bodyId     type             instance
0       200326126    CL141              CL141_L
1       202916528     FS4C      FS4C(FQ12)_C4_R
2       203253072     FS4A  FS4A(AB)(FQ12)_C3_R
3       203253253   SMP505      SMP505(PDM22)_L
4       203257652     FS4C      FS4C(FQ12)_C6_R
...           ...      ...                  ...
21734  7112579856     LC20                 LC20
21735  7112615127  AVLP335            AVLP335_R
21736  7112617294  AVLP331            AVLP331_R
21737  7112622044   LAL137      LAL137(PVL05)_L
21738  7112622236      NaN                  NaN

[21739 rows x 3 columns]

In [3]:
df_roi_connections = pd.read_csv('traced-roi-connections.csv')
df_roi_connections

bodyId_pre  bodyId_post     roi  weight
0         200326126    264083994  SLP(R)       3
1         200326126    295816140  ICL(R)       2
2         200326126    295816140  SCL(R)       2
3         200326126    295816140  PLP(R)       1
4         200326126    296203440  SCL(R)       1
...             ...          ...     ...     ...
4259619  7112622236    328283521  SMP(R)       1
4259620  7112622236    357932060  SMP(R)       1
4259621  7112622236    357940977  SMP(R)       1
4259622  7112622236    358631450  SMP(R)       1
4259623  7112622236    480672667  SLP(R)       1

[4259624 rows x 4 columns]

In [4]:
df_connections = pd.read_csv('traced-total-connections.csv')
df_connections

bodyId_pre  bodyId_post  weight
0         294437328    295470623       1
1         294437328    295133902       1
2         294437328    448260940       1
3         294437328    294783423       1
4         294437328   5812979995       1
...             ...          ...     ...
3550398  5901222127    546307807       1
3550399  5901222127    577697202       1
3550400  5901222127    393210919       1
3550401  5901222127    330640044       5
3550402  5901222127    516875221       3

[3550403 rows x 3 columns]

# Analysis

In [5]:
len(df_neurons)

21739

In [6]:
df_roi_connections['roi'].unique()

array(['SLP(R)', 'ICL(R)', 'SCL(R)', 'PLP(R)', 'AVLP(R)', 'LH(R)',
       'SMP(L)', 'FB', 'AB(R)', 'SMP(R)', 'SIP(R)', 'SIP(L)', 'SCL(L)',
       'NotPrimary', 'SPS(L)', 'IPS(R)', 'SPS(R)', 'BU(L)', 'CRE(L)',
       'IB', 'AME(R)', 'CA(R)', 'FLA(R)', 'CRE(R)', 'EPA(R)', 'bL(R)',
       "b'L(R)", 'bL(L)', 'PED(R)', 'aL(R)', 'gL(R)', "b'L(L)", 'gL(L)',
       'ICL(L)', 'AB(L)', 'AL(R)', 'ATL(L)', 'LAL(R)', 'EB', 'PVLP(R)',
       'ATL(R)', 'AOTU(R)', "a'L(R)", 'LO(R)', 'VES(R)', 'aL(L)',
       'GOR(R)', 'WED(R)', 'LAL(L)', 'NO', 'PB', 'SAD', 'BU(R)', "a'L(L)",
       'GOR(L)', 'GNG', 'CAN(R)', 'VES(L)', 'EPA(L)', 'AL(L)', 'ME(R)',
       'LOP(R)'], dtype=object)

In [7]:
df_neurons['type'].unique()

array(['CL141', 'FS4C', 'FS4A', ..., 'aMe6b', 'AVLP534', 'SLP169'],
      shape=(5555,), dtype=object)

In [8]:
l = [x for x in df_neurons['type'].unique() if 'KC' in str(x)]
l

['KCab-m',
 'KCab-c',
 'KCab-s',
 'KCg-m',
 'KCab-p',
 "KCa'b'-ap2",
 "KCa'b'-m",
 "KCa'b'-ap1",
 'KCg-d',
 'KCg-t',
 'KCg-s4',
 'KCg-s1',
 'KCg-s3',
 'KCg-s2']

In [9]:
KC_types = ['KCab-m',
 'KCab-c',
 'KCab-s',
 'KCg-m',
 'KCab-p',
 "KCa'b'-ap2",
 "KCa'b'-m",
 "KCa'b'-ap1",
 'KCg-d',
 'KCg-t',
 'KCg-s4',
 'KCg-s1',
 'KCg-s3',
 'KCg-s2']

In [10]:
l = [x for x in df_neurons['type'].unique() if 'APL' in str(x)]
l

['APL']

In [11]:
APL_type = ['APL']

# Filtering

today try doing the filtering of neurons by regions

In [12]:
df_neurons

bodyId     type             instance
0       200326126    CL141              CL141_L
1       202916528     FS4C      FS4C(FQ12)_C4_R
2       203253072     FS4A  FS4A(AB)(FQ12)_C3_R
3       203253253   SMP505      SMP505(PDM22)_L
4       203257652     FS4C      FS4C(FQ12)_C6_R
...           ...      ...                  ...
21734  7112579856     LC20                 LC20
21735  7112615127  AVLP335            AVLP335_R
21736  7112617294  AVLP331            AVLP331_R
21737  7112622044   LAL137      LAL137(PVL05)_L
21738  7112622236      NaN                  NaN

[21739 rows x 3 columns]

In [15]:
kc_neurons = df_neurons[df_neurons['type'].isin(KC_types)]
kc_neurons

bodyId    type  instance
484     300968622  KCab-m  KCab-m_R
487     301309622  KCab-c  KCab-c_R
488     301314150  KCab-c  KCab-c_R
489     301314154  KCab-c  KCab-c_R
490     301314208  KCab-c  KCab-c_R
...           ...     ...       ...
21607  5901202076  KCab-c  KCab-c_R
21608  5901202102   KCg-m   KCg-m_R
21635  5901205770  KCab-s  KCab-s_R
21642  5901207528  KCab-c  KCab-c_R
21719  5901225361   KCg-m   KCg-m_R

[1927 rows x 3 columns]

In [16]:
kc_neurons['instance'].unique()

array(['KCab-m_R', 'KCab-c_R', 'KCab-s_R', 'KCg-m_R', 'KCab-p_R',
       "KCa'b'-ap2_R", "KCa'b'-m_R", "KCa'b'-ap1_R", 'KCg-d_R', 'KCg-t_R',
       'KCg-s4_R', 'KCg-s1(super)_R', 'KCg-s3_R', 'KCg-s2(super)_R'],
      dtype=object)

In [22]:
# IDs of pre_KC neurons
pre_kc_ids = df_connections[df_connections['bodyId_post'].isin(kc_neurons['bodyId'])]['bodyId_pre'].to_list()
pre_kc_ids

[295461570,
 295461570,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268725517,
 268

In [26]:
# all pre KC neurons
pre_kc_neurons = df_neurons[df_neurons['bodyId'].isin(pre_kc_ids)]
pre_kc_neurons

bodyId     type         instance
0       200326126    CL141          CL141_L
34      264083994     DN1a           DN1a_R
84      268725517    PAM09   PAM09(B1ped)_L
85      268731005   SMP146  SMP146(ADM10)_L
92      294436967   PPL203         PPL203_R
...           ...      ...              ...
21659  5901213440   CRE011         CRE011_R
21663  5901213816  LHPV5j1        LHPV5j1_R
21690  5901217539     LC34           LC34_R
21711  5901222910  DM2_lPN        DM2_lPN_R
21719  5901225361    KCg-m          KCg-m_R

[3144 rows x 3 columns]

In [28]:
pre_kc_neurons['instance'].unique()

array(['CL141_L', 'DN1a_R', 'PAM09(B1ped)_L', 'SMP146(ADM10)_L',
       'PPL203_R', 'SLP371_R', 'SLP359_R', 'SLP269_R', 'SMP357_R',
       'SMP432_a_R', 'LHPV6f3_b_R', 'SLP257_R', 'PAM10(B1)_R', 'SMP146_R',
       'PAM09(B1ped)_R', 'KCab-m_R', 'MBON14(a3)_R', 'KCab-c_R',
       'SLP446_R', 'SLP367_R', 'SLP272_R', 'SLP457(SCB013)_R', 'SLP207_R',
       'SMP362_R', 'SLP069_R', 'LHCENT2_R', 'SMP434_R', 'SLP142_R',
       'PPL201_R', 'LHCENT1_R', 'PAM10(B1)_L', 'OA-VPM3(NO2/NO3)_R',
       'SLP028_R', 'SMP568_d_R', 'LHCENT9_R', 'PPL106(a3)_R', 'KCab-s_R',
       'KCg-m_R', 'LHPV5h3_R', 'SLP445_R', 'LHPD3b1_b_R', 'SLP305_R',
       'LHPV4l1_R', 'SLP436_R', 'LHPV5l1_R', 'SLP365_R', 'SLP358_R',
       'SLP295_R', 'SMP239_R', 'SLP306_R', 'LHPV6b1_R', 'LHCENT8_R',
       'AstA1_L', nan, 'SMP529_R', 'LHPV6i1_a_R', 'SLP308_R',
       'LHPV6h1_a_R', 'LHPV6c1_R', 'SLP462(SCB070)_R', 'SLP285_R',
       'CL194_R', 'SLP466(SCB067)_R', 'LHPV6m1_R', 'SLP366_R', 'SLP360_R',
       'SMP531_R', 'LHPV5h2_R'

# Lets try another way

lets try keeping only the Kenyon Cells and recurrent connections between them.

In [31]:
kc_connections = df_connections[
    df_connections['bodyId_post'].isin(kc_neurons['bodyId']) &
    df_connections['bodyId_pre'].isin(kc_neurons['bodyId'])
]
kc_connections

bodyId_pre  bodyId_post  weight
67432     301314154    549576424       1
67433     301314154    736126717       2
67435     301314154    829568160       1
67437     301314154   5812983309       1
67438     301314154   5812983575       1
...             ...          ...     ...
3547238  5901202076    394755376       1
3547239  5901202076   5812982760       4
3547240  5901202076    394755632       1
3547242  5901202076   5813022050       6
3547244  5901202076    477262101       5

[340367 rows x 3 columns]

In [36]:
len(kc_connections['bodyId_pre'].unique())

1927

In [37]:
len(kc_connections['bodyId_post'].unique())

1927

In [32]:
kc_neurons

bodyId    type  instance
484     300968622  KCab-m  KCab-m_R
487     301309622  KCab-c  KCab-c_R
488     301314150  KCab-c  KCab-c_R
489     301314154  KCab-c  KCab-c_R
490     301314208  KCab-c  KCab-c_R
...           ...     ...       ...
21607  5901202076  KCab-c  KCab-c_R
21608  5901202102   KCg-m   KCg-m_R
21635  5901205770  KCab-s  KCab-s_R
21642  5901207528  KCab-c  KCab-c_R
21719  5901225361   KCg-m   KCg-m_R

[1927 rows x 3 columns]

so now we have all 1927 neurons in the Right MB of Drosophila, and the 340367 connections between pairs of these

what i need next is to create a network model of this hemispheric MB. then inputs and APL and ouputs can be generic